lang/cs/Org.Apache.REEF.Common/Runtime/Evaluator/Task/TaskRuntime.cs

```diff
@@ -133,25 +133,11 @@ public void Start()
                     var e = new InvalidOperationException("TaskRuntime not in Running state, instead it is in state " + _currentStatus.State);
                     Org.Apache.REEF.Utilities.Diagnostics.Exceptions.Throw(e, LOGGER);
                 }
-                byte[] result;
+
                 byte[] taskMemento = _memento.IsPresent() ? _memento.Value : null;
-                System.Threading.Tasks.Task<byte[]> runTask = new System.Threading.Tasks.Task<byte[]>(() => RunTask(taskMemento));
-                try
-                {
-                    runTask.Start();
-                    runTask.Wait();
-                }
-                catch (Exception e)
-                {
-                    Org.Apache.REEF.Utilities.Diagnostics.Exceptions.CaughtAndThrow(e, Level.Error, "Exception thrown during task running.", LOGGER);
-                }
-                result = runTask.Result;
+                var result = RunTask(taskMemento);
```

Comment:
Couple of things. 
t = new Task<>();
t.Start();

is not used.
Task.Run(); or (not recommended) Task.Factory.StartNew are the canonical ways of starting a task.
Secondly, calling Wait() was anyways blocking the thread, but since we have used Task.Run(() => taskRuntime.Start()); in ContextRuntime we are fine (may be).